models/shopify__daily_shop.sql

```diff
@@ -0,0 +1,213 @@
+with shop as (
+
+    select *
+    from {{ var('shopify_shop') }}
+),
+
+calendar as (
+
+    select *
+    from {{ ref('shopify__calendar') }}
+    where cast({{ dbt.date_trunc('day','date_day') }} as date) = date_day
+
+),
+
+shop_calendar as (
+
+    select
+        calendar.date_day,
+        shop.*
+
+    from calendar
+    join shop on
+        cast(shop.created_at as date) <= calendar.date_day
+),
+
+orders as (
+
+    select *
+    from {{ ref('shopify__orders') }}
+
+    where not coalesce(is_deleted, false)
+),
+
+order_aggregates as (
+
+    select
+        source_relation,
+        cast({{ dbt.date_trunc('day','created_timestamp') }} as date) as date_day,
+        count(distinct order_id) as count_orders,
+        sum(line_item_count) as count_line_items,
+        count(distinct customer_id) as count_customers,
+        count(distinct email) as count_customer_emails,
+        sum(order_adjusted_total) as order_adjusted_total,
+        avg(order_adjusted_total) as avg_order_value,
+        sum(shipping_cost) as shipping_cost,
+        sum(order_adjustment_amount) as order_adjustment_amount,
+        sum(order_adjustment_tax_amount) as order_adjustment_tax_amount,
+        sum(refund_subtotal) as refund_subtotal,
+        sum(refund_total_tax) as refund_total_tax,
+        sum(total_discounts) as total_discounts,
+        sum(shipping_discount_amount) as shipping_discount_amount,
+        sum(percentage_calc_discount_amount) as percentage_calc_discount_amount,
+        sum(fixed_amount_discount_amount) as fixed_amount_discount_amount,
+        sum(count_discount_codes_applied) as count_discount_codes_applied,
+        count(distinct location_id) as count_locations_ordered_from,
+        sum(case when count_discount_codes_applied > 0 then 1 else 0 end) as count_orders_with_discounts,
+        sum(case when refund_subtotal > 0 then 1 else 0 end) as count_orders_with_refunds,
+        min(created_timestamp) as first_order_timestamp,
+        max(created_timestamp) as last_order_timestamp
+
+    from orders
+    group by 1,2
+
+),
+
+order_lines as(
+
+    select *
+    from {{ ref('shopify__order_lines') }}
+),
+
+order_line_aggregates as (
+
+    select
+        order_lines.source_relation,
+        cast({{ dbt.date_trunc('day','orders.created_timestamp') }} as date) as date_day,
+        sum(order_lines.quantity) as quantity_sold,
+        sum(order_lines.refunded_quantity) as quantity_refunded,
+        sum(order_lines.quantity_net_refunds) as quantity_net,
+        count(distinct order_lines.variant_id) as count_variants_sold, 
+        count(distinct order_lines.product_id) as count_products_sold, 
+        sum(case when order_lines.is_gift_card then order_lines.quantity_net_refunds else 0 end) as quantity_gift_cards_sold,
+        sum(case when order_lines.is_shipping_required then order_lines.quantity_net_refunds else 0 end) as quantity_requiring_shipping
+
+    from order_lines
+    join orders
```

Comment:
Should this be left?

Reply:
all order lines should definitely have orders, but i see the appeal of doing left joins by default

Reply:
changed it to left